src/analysis/ppx_expand.ml

```diff
@@ -99,9 +99,13 @@ let get_ppxed_source ~ppxed_parsetree ~pos ppx_kind_with_attr :
     | Sig_item original_sg, _ -> (
         let signature_item (self : Ast_iterator.iterator)
             (new_sg : Parsetree.signature_item) =
-          match check_at_pos pos new_sg.psig_loc && original_sg <> new_sg with
-          | true -> signature := new_sg :: !signature
-          | false -> Ast_iterator.default_iterator.signature_item self new_sg
+          let included =
+            Location_aux.included new_sg.psig_loc ~into:original_sg.psig_loc
+          in
+          match included && original_sg <> new_sg, new_sg.psig_loc.loc_ghost with
+          | true, _ -> signature := new_sg :: !signature
```

Comment:
This works great for `signature_items` but for `structure_items` something weird happens, the result looks like this:
```
;;rename
 include
    struct
      let _ = fun (_ : point) -> ()
      type point_renamed = {
        x: int ;
        y: int }
    end[@@ocaml.doc "@inline"][@@merlin.hide ]
```
There is an extra `;;rename` added which can be removed if we enforce `ghost_locs` for the `structure_items`